test/e2e/webhook_test.go

```diff
@@ -1366,3 +1367,216 @@ var _ = Describe("webhook tests for ClusterResourcePlacementEviction CREATE oper
+				Type:   intstr.String,
+				StrVal: "50%",
+			}
+			By(fmt.Sprintf("expecting denial of UPDATE disruption budget %s", crpName))
+			err := hubClient.Update(ctx, &crpdb)
+			if k8sErrors.IsConflict(err) {
+				return err
+			}
+			var statusErr *k8sErrors.StatusError
+			Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Update CRPDB call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8sErrors.StatusError{})))
+			Expect(statusErr.Status().Message).Should(MatchRegexp(fmt.Sprintf("cluster resource placement policy type PickAll is not supported with min available as a percentage %v", crpdb.Spec.MinAvailable)))
+			return nil
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+	})
+
+	It("should deny update on CRPDB with MaxUnavailable as percentage and PickAll CRP", func() {
+		// Update the CRPDB.
+		Eventually(func(g Gomega) error {
+			var crpdb placementv1beta1.ClusterResourcePlacementDisruptionBudget
+			g.Expect(hubClient.Get(ctx, types.NamespacedName{Name: crpName}, &crpdb)).Should(Succeed(), "Failed to get CRPDB %s", crpName)
+			crpdb.Spec.MaxUnavailable = &intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "75%",
+			}
+			crpdb.Spec.MinAvailable = nil // Clear MinAvailable to test MaxUnavailable
+			By(fmt.Sprintf("expecting denial of UPDATE disruption budget %s", crpName))
+			err := hubClient.Update(ctx, &crpdb)
+			if k8sErrors.IsConflict(err) {
+				return err
+			}
+			var statusErr *k8sErrors.StatusError
+			Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Update CRPDB call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8sErrors.StatusError{})))
+			Expect(statusErr.Status().Message).Should(MatchRegexp(fmt.Sprintf("cluster resource placement policy type PickAll is not supported with any specified max unavailable %v", crpdb.Spec.MaxUnavailable)))
+			return nil
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+	})
+
+	It("should deny update on CRPDB with MaxUnavailable as integer and PickAll CRP", func() {
+		// Update the CRPDB.
+		Eventually(func(g Gomega) error {
+			var crpdb placementv1beta1.ClusterResourcePlacementDisruptionBudget
+			g.Expect(hubClient.Get(ctx, types.NamespacedName{Name: crpName}, &crpdb)).Should(Succeed(), "Failed to get CRPDB %s", crpName)
+			crpdb.Spec.MaxUnavailable = &intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 1,
+			}
+			crpdb.Spec.MinAvailable = nil // Clear MinAvailable to test MaxUnavailable
+			By(fmt.Sprintf("expecting denial of UPDATE disruption budget %s", crpName))
+			err := hubClient.Update(ctx, &crpdb)
+			if k8sErrors.IsConflict(err) {
+				return err
+			}
+			var statusErr *k8sErrors.StatusError
+			Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Update CRPDB call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8sErrors.StatusError{})))
```

Comment:
https://onsi.github.io/ginkgo/#testing-external-systems, need to use g.Expect or else Eventually won't retry,

From link attached,
"You may be wondering why we need to pass in a dedicated Gomega instance to the polled function. That's because the default global-level assertions are implicitly tied to Ginkgo's Fail handler. The first failed assertion in an Eventually would cause the spec to fail with no possibility to retry. By passing in a fresh Gomega instance, Eventually can monitor for failures itself and control the final failure/success state of the assertion it is governing."